core/src/autogluon/core/scheduler/seq_scheduler.py

```diff
@@ -0,0 +1,147 @@
+
+        self.training_history = training_history
+        if trial not in self.training_history:
+            self.training_history[trial] = []
+
+        self.task_config = EasyDict(deepcopy(config))
+
+        self.config_history = config_history
+
+        self.trial_started = time.time()
+        self.last_reported_time = self.trial_started
+        self.last_result = None
+
+    def __call__(self, *args, **kwargs):
+        result = deepcopy(kwargs)
+
+        if 'done' not in result:
+            result['trial'] = self.trial
+
+            now = time.time()
+            result['time_this_iter'] = now - self.last_reported_time
+            result['time_since_start'] = now - self.trial_started
+            self.last_reported_time = now
+
+            self.training_history[self.trial].append(result)
+
+            if self.trial not in self.config_history:
+                self.config_history[self.trial] = self.task_config
+                if 'util_args' in self.task_config:
+                    self.task_config.pop('util_args')
+
+            self.last_result = result
+        if 'traceback' in result:
+            self.searcher.evaluation_failed(config=self.task_config, **result)
+
+    def terminate(self):
+        pass  # compatibility
+
+
+class LocalSequentialScheduler(object):
+
+    def __init__(self, train_fn, reward_attr, time_attr='epoch', searcher='auto', **kwargs):
+        self.train_fn = train_fn
+        self.training_history = None
+        self.config_history = None
+        self._reward_attr = reward_attr
+        self.time_attr = time_attr
+        self.resource = kwargs['resource']
+        self.max_reward = kwargs.get('max_reward')
+
+        if searcher is 'local_sequential_auto':
+            searcher = 'auto'
+
+        if searcher is 'auto':
+            searcher = GPFIFOSearcher
+        self.searcher = searcher(self.train_fn.cs, reward_attribute=self._reward_attr)
+
+        self.num_trials = kwargs.get('num_trials', 9999)
+        self.time_out = kwargs.get('time_out')
+        if self.num_trials is None:
+            assert self.time_out is not None, \
+                "Need stopping criterion: Either num_trials or time_out"
+
+        self.metadata = {
+            'search_space': train_fn.kwspaces,
+            'search_strategy': searcher,
+            'stop_criterion': {
+                'time_limits': self.time_out,
+                'max_reward': self.max_reward},
+            'resources_per_trial': self.resource}
+
+    def run(self, **kwargs):
+        self.training_history = OrderedDict()
+        self.config_history = OrderedDict()
+
+        time_start = time.time()
+        for i in tqdm(range(self.num_trials)):
+            self.run_trial(task_id=i)
+            time_end = time.time()
+            if self.time_out is not None:
+                if time_end - time_start >= self.time_out:
+                    logger.log(20, f'\tTime limit exceeded...')
+                    break
+
```

Comment:
Missing the run_with_config method

Reply:
Tabular currently don't use this method.